src/activity_item_handling.cpp

```diff
@@ -1317,18 +1317,17 @@ static activity_reason_info can_do_activity_there( const activity_id &act, Chara
                     // we need a shovel/hoe
                     return activity_reason_info::fail( do_activity_reason::NEEDS_TILLING );
                 }
+                // do we have the required seed on our person?
+                // If its a farm zone with no specified seed, and we've checked for tilling and harvesting.
+                // then it means no further work can be done here
+            } else if( seed.is_empty() ) {
+                return activity_reason_info::fail( do_activity_reason::ALREADY_DONE );
             } else if( here.has_flag_ter_or_furn( seed->seed->required_terrain_flag, src_loc ) &&
                        // TODO: fix point types
                        warm_enough_to_plant( src_loc.raw() ) ) {
```

Comment:
One way to fix the clang-tidy error is to merge this "seed is empty" block with the else block at line 1341
```suggestion
            } else if( here.has_flag_ter_or_furn( seed->seed->required_terrain_flag, src_loc ) &&
                       // TODO: fix point types
                       warm_enough_to_plant( src_loc.raw() ) &&
                       // do we have the required seed on our person?
                       // If its a farm zone with no specified seed, and we've checked for tilling and harvesting.
                       // then it means no further work can be done here
                       !seed.is_empty() ) {
```

Reply:
I am worried that allowing has_flag_ter_or_furn to even check will cause the segfault. Based on reading the clang error, it seems clang wants me to turn

`            } else if( seed.is_empty() ) {`

into

`            }
            else if( seed.is_empty() ) {`

This would require me to edit substantial portions of the proc, since most of the proc does this.